### DISCUSSION TITLE
Annotation Processing : How does it work ?

### DISCUSSION THREAD

**Discussion by Camille-Racine** [JacobBumgarner/VesselVio#67] [General]
Hi,
Thank you for sharing your code and the VesselVio Software. It's a pleasure to try a specific software for vasculature. 
I've just started using it, and some features are not so easy to understand for me. 
I'm working on the vascular networks during the development and I've understood how to visualize or analyze the vessels. 
The vessels have been modelized on Imaris, then binarized and convert to .tiff or .nii formats with fiji. But, I would like to use the annotation processing option. The aim would be to register my networks to Allen developing mouse brain atlas. 
According to your "star methods" publication, I'm pretty sure that your software could be the best way to do this registration. Unfortunaly, I don't understand how it works, the meaning of "annotation file" which must be in .nii formats, how to use the .json files when I export the ROI ... ?
Could you please, give me a tutorial to better understand this feature ?  
Thank you in advance

**Comment by JacobBumgarner** [comment]
Hi there,
Thanks for reaching out. VesselVio does not provide annotation functionality. 

If you're hoping to register your mouse brain dataset to the Allen atlas, you'll need to use registration software such as QuickNII, or see the Allen atlas publication description of registration solutions [here](https://www.sciencedirect.com/science/article/pii/S0092867420304025#:~:text=many%20recent%20publications).

You can then load in the annotation files alongside your segmented vasculature dataset and specify which regions you'd like to visualize and/or analyze.

**Reply by Camille-Racine** [reply]
Hi, 
Thanks for the quick reply! 
Ok, I see ! I'll try these solutions and will go back to your software.

**Comment by Camille-Racine** [comment]
Hi, 
In the same topic, I've got a question. I've just obtained an Atlas segmentation created using itk-SNAP. First I would like to know if it's "now" possible to import my segmentation in VesselVio. The segmentation file in .nii, the label file is in LABEL but can be convert in .json. 
In fact when I try to load my own label file I received an "error loading tree file". But I'm sure that all is typed correctly ... 
So, do you think that I could import my own segmentation and the corresponding label in the software ? 
How can I solve this problem ? DO you have an example of file which could be load in the "Load json" ? 
Thank you in advance.

**Reply by JacobBumgarner** [reply]
Hi @Camille-Racine , you will need to create a `JSON` file using the "Annotation Processing" page. 

I'm assuming that your `.nii` segmentation file has some `int` data type.

On the Annotation Processing page, you'll click "New ROI", define the `int` that characterizes a specific region, and then define whatever color you'd like to see that region annotated by.

For example, imagine the image:
```
0 0 0 1 0 
0 0 1 1 0 
0 0 1 1 1 
```

With the corresponding annotation:
```
0 0 0 1 0 
0 0 3 1 0 
0 0 3 2 2
```

This annotation has three separate regions, indicated by the values `1`, `2`, and `3`.

To analyze these regions, you'll go into the Annotation processing page, add a "New ROI" for each region number and give each region a color:
- Region 1: `ID = 1`, `Color = "#FF0000"`
- Region 2: `ID = 2`, `Color = "#00FF00"`
- Region 3: `ID = 3`, `Color = "#0000FF"`

You will then export this annotation file (a `json` file) and load it with your image/annotation.

I hope this explanation was helpful and clear. Please let me know if you have any trouble/questions.

**Reply by JacobBumgarner** [reply]
If the annotation type you've created from itk-SNAP used the same annotations as the developing mouse brain annotations already stored in VesselVio, you could instead just select the regions that you want to analyze in the left menu, add them to the right menu, and export them that way.

You should definitely confirm that the annotation numbers match the atlas you used for itk-SNAP and the annotation values stored in VesselVio from the Allen Institue.

**Comment by Camille-Racine** [comment]
Hi ! 
First,  thank you for your reply. Unfortunately, it still doesn't work. 
I've sent you a private message on your email address (the one linked with your Github). Did you receive it ? 
Otherwise, could you give me a way to contact you in private ? We could send you the files we are trying to open in the software.
Have a good day !

**Reply by JacobBumgarner** [reply]
I'm sorry to hear that, I'm not sure what is going on!

I'll have to update that email address, thanks for the heads up. You can contact me via jacobbum21 at gmail.
